### DISCUSSION TITLE
Release v1.0.0

### DISCUSSION THREAD

**Discussion by JensenNick** [JensenNick/huawei_solar_pees#24] [Announcements]
## Release Note / v1.0.0
The *"Huawei Solar PEES package file"* has now been tested on a clean install of Home Assistant - no strings to any thing in the past. This is hopefully how most of you will experience the "installation".

**New naming convention:**
This wasn't an easy decision, but for the better I believe looking forward. Template sensors lose their value temporarily while HA is rebooting / restarting, leaving a blank space in the graphs. I therefor upgraded the *"Huawei Solar PEES package"* with utility meters for all the platform and template sensors.

I could have gone with a new suffix some of the utility meters, but that would have been inconsistent and an annoyance when these utility meters would be used in cards, energy dashboard etc. - you would have had to change the name of the source every time you use it. So, I have decided to re-name the platform sensors, template sensors and template trigger sensors respectively with the suffix as follows,
* `_ps` for platform sensor,
* `_ts` for template sensor and 
* `_tt` for template trigger sensor.

There are other bonus effects to this as well, 1) utility meters are easier to reset and calibrate without the need for a deep dive into your database 2) you can now easily identify the origin of the sensor and 3) easy filtering based on the new suffix.

**Error in code:** 
A crucial error in the attribute for the sensor `battery_charge_yield_sale` has been corrected (new sensor name `battery_charge_yield_sale_tt`),

**PV template sensors:**
The device_class on the PV sensors has been removed, in order to allow state_ class: total_increasing. Forum posts from the Developers suggest not to use device_class for this type of sensors.

**Dashboards:**
Codes for input to the dashboards have ben updated so they match the new naming convention.

**Info:**
Version number, domain name and code owner has been added to the `huawei_solar_pees.yaml` package file

**Documentation:**
~Has not yet been updated accordingly to above changes.~
*As of 7th of February 2024, the documentation has been updated.*

**Full Changelog**: https://github.com/JensenNick/huawei_solar_pees/commits/v1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/JensenNick/huawei_solar_pees/releases/tag/v1.0.0'>Release v1.0.0</a>.</em>

**Comment by JensenNick** [comment]
I am not sure, but I think it may be necessary to Restart Home Assistant twice after "installing" the package file. I had to do yet another clean install and I struggled to get the energy values to show. I would appreciate if anyone could confirm this before I correct the README.

I think the reason is that the template trigger sensors need to populate a `last` and maybe also a `lastlast` value on order for them to initiate calculations.
![2024-02-03_10-47-06](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/ea10e587-8988-44bf-bb6a-a5cf3c25c6e8)

**Reply by JacobsenKim** [reply]
jeg har lige lavet en ny install.

nogle af mine _ts samt _tt siger også unavailable, unknown, så søger jeg på _ps og sætter alle der ikke viser tal til 0, så når jeg søger på _ts samt _tt så er de alle med tal.

Alt kører i dashboard og Economy - NRI PV tæller.

![Skærmbillede fra 2024-02-03 15-36-42](https://github.com/JensenNick/huawei_solar_pees/assets/157890151/1938c6d5-d6ea-4e4a-a10d-c7daf1c74757)

**Reply by JensenNick** [reply]
> nogle af mine _ts samt _tt siger også unavailable, unknown, så søger jeg på _ps og sætter alle der ikke viser tal til 0, så når jeg søger på _ts samt _tt så er de alle med tal.

@JacobsenKim Prøvede du at genstarte og refresh browser to gange?

**Reply by JacobsenKim** [reply]
Ja billede er efter restart og med Ctrl + F5, har genstartet flere gange efter alt kører.

**Reply by JensenNick** [reply]
@JacobsenKim Jeg er faldet over en fejl som muligvis har indflydelse på hvorfor sensorerne ikke starter op. Jeg har lavet en branchen *"[corrected_tt_sensors](https://github.com/JensenNick/huawei_solar_pees/tree/correct_tt_sensors)"* med rettelserne til `huawei_solar_pees.yaml` filen.

**Reply by JensenNick** [reply]
Det ser ikke ud til at ovenstående ændrer noget. Power sensorer returnerer "0", men energy sensorerne returnerer "unknown". Testet på ny install af HA dog uden Huawei Solar integrationen. Har ændret branch navn til v1.0.1 og så samler jeg rettelser her.

**Comment by JensenNick** [comment]
I get the "FIX ISSUE" message on all utility meters in *"Developer tools > STATISTICS"*. They dont seem to pick up the `unit_of_measurement` from the sensor. I dont se it as an issue (yet) but it is admittedly annoying.

![2024-02-03_14-49-12](https://github.com/JensenNick/huawei_solar_pees/assets/149193835/3e26110f-850d-43aa-952c-518400a85256)

**Reply by JacobsenKim** [reply]
Hvis der går forlang tid før der er aktivitet så sætter den ikke unit_of_measurement, det var noget jeg bøvlede med den gang jeg selv var igang.
